packages/lib/src/components/internal/Address/components/AddressSearch.tsx

```diff
@@ -0,0 +1,98 @@
+import Field from '../../FormFields/Field';
+import { Fragment, h } from 'preact';
+import renderFormField from '../../FormFields';
+import { AddressLookupItem } from '../types';
+import { useCallback, useEffect, useMemo, useState } from 'preact/hooks';
+import './AddressSearch.scss';
+import useCoreContext from '../../../../core/Context/useCoreContext';
+import { debounce } from '../utils';
+
+export type OnAddressLookupType = (
+    value: string,
+    actions: {
+        resolve: (value: Array<AddressLookupItem>) => void;
+        reject: (reason?: any) => void;
+    }
+) => Promise<void>;
+
+interface AddressSearchProps {
+    onAddressLookup?: OnAddressLookupType;
+    onSelect: any; //TODO
+    onManualAddress: any;
+    externalErrorMessage: string;
+    hideManualButton: boolean;
+}
+
+export default function AddressSearch({ onAddressLookup, onSelect, onManualAddress, externalErrorMessage, hideManualButton }: AddressSearchProps) {
+    const [formattedData, setFormattedData] = useState([]);
+    const [originalData, setOriginalData] = useState([]);
+
+    const [errorMessage, setErrorMessage] = useState('');
+
+    const { i18n } = useCoreContext();
+    const mapDataToSelect = data => data.map(({ id, name }) => ({ id, name }));
+
+    const onInput = useCallback(
+        async event => {
```

Comment:
Is this an actual Input `event` or `value`? Seems we pass the value but by looking it here looks like we pass the event itself. Adding proper types would help to understand the code